src/hotspot/share/gc/g1/g1FullGCPrepareTask.cpp

```diff
@@ -67,6 +67,15 @@ bool G1FullGCPrepareTask::G1CalculatePointersClosure::do_heap_region(HeapRegion*
       // Force the high live ration region pinned,
       // as we need skip these regions in the later compact step.
       force_pinned = true;
+      if (hr->is_young()) {
+        // Old regions have BOTs info for performance consideration, but young regions
```

Comment:
I'd replace this comment with something like this:

    // G1 updates the BOT for old region contents incrementally, but young regions lack BOT 
    // information for performance reasons.
    // Recreate BOT information of high live ratio young regions here to keep expected
    // performance during scanning their card tables in the collection pauses later.

Or something like this...

Reply:
Thanks Thomas for refining the comments.